### PR TITLE
make country level chart layout better

### PR DESCRIPTION
**Pull request by drifter089**
image

### CODE REVIEW

**Review by BryanFran — APPROVED** [review]
Great Akshat!
I'll just recommend making charts size fit in view. It looks great!

**Comment by drifter089** [comment]
@BryanFran  and @ekatef  ,
i will try to stack carriers and make three columns in bar chart then we can do a revision on layout

**Review by yerbol-akhmetov — COMMENTED** [review]
I have left comments above.

**Comment by drifter089** [comment]
got it will make three columns then make it as compact as possible

**Comment by drifter089** [comment]
image

**Comment by drifter089** [comment]
image

**Comment by yerbol-akhmetov** [comment]
> image

I think we can make bars even more thinner? What do you think @BryanFran ?

File: components/Charts/GenerationMixBarChartStacked.tsx [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    if (data?.current?.data) {
+      let dataArray = Array.isArray(data.current.data) ? data.current.data : [];
+      console.log("dataArray", dataArray);
+
+      dataArray = dataArray.filter(
+        (item: any) =>
+          item &&
+          item.carrier &&
+          item.carrier !== "Total capacity" &&
+          item.carrier !== "Geothermal"
```
Comment (ekatef) [inline components/Charts/GenerationMixBarChartStacked.tsx:131]: Are there specific reasons to exclude geothermal?

File: components/Charts/GenerationMixBarChartStacked.tsx [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+          geothermal: 0,
+          loadShedding: 0,
+          totalGeneration: 0,
+        },
+        {
+          model: "PyPSA",
+          biomass: 0,
+          coal: 0,
+          oil: 0,
+          naturalGas: 0,
```
Comment (ekatef) [inline components/Charts/GenerationMixBarChartStacked.tsx:155]: The revision looks good 😄

File: components/Charts/GenerationMixBarChartStacked.tsx [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+type CarrierKey = (typeof carrierMap)[keyof typeof carrierMap];
+
+const chartConfig = {
+  biomass: {
+    label: "Biomass",
+    color: "hsl(var(--chart-1))",
+  },
+  coal: {
+    label: "Coal",
+    color: "hsl(var(--chart-2))",
```
[the remark this answers is not in the document]
Comment (yerbol-akhmetov) [reply components/Charts/GenerationMixBarChartStacked.tsx:74]: Yes, coal and lignite can be grouped to coal.

**Pull request merged**